packages/react-keyring/src/providers/Keyring.tsx

```
@@ -129,62 +125,65 @@ export function KeyringProvider ({
   }
 
   const createSpace = async (name?: string): Promise<DID> => {
-    const agent = await getAgent()
-    const { did } = await agent.createSpace(name)
-    await agent.setCurrentSpace(did)
-    setSpace(getCurrentSpaceInAgent(agent))
+    const c = await getClient()
+    const space = await c.createSpace(name ?? 'Unnamed Space')
+    const did = space.did()
+    await c.setCurrentSpace(did)
+    setSpace(c.currentSpace())
     return did
   }
 
-  const registerSpace = async (email: string, opts: RegisterSpaceOpts = {}): Promise<void> => {
-    const agent = await getAgent()
-    await agent.registerSpace(email, {
-      provider: opts.provider ?? (agent.connection.id.did() as DID<'web'>)
-    })
-    setSpace(getCurrentSpaceInAgent(agent))
-    setSpaces(getSpaces(agent))
+  const registerSpace = async (email: string): Promise<void> => {
+    const c = await getClient()
+    const account = useAccount(c, { email })
+    const space = c.currentSpace()
+    if (account && space) {
+      await account.provision(space.did() as DIDKey)
+      setSpace(c.currentSpace())
+      setSpaces(c.spaces())
+    }
```

Need to throw if missing account or space.